packages/starlight/utils/navigation.ts

```diff
@@ -134,8 +141,15 @@ function makeLink(
 	badge?: Badge,
 	attrs?: LinkHTMLAttributes
 ): Link {
-	if (!isAbsolute(href)) href = pathWithBase(href);
-	const isCurrent = href === ensureTrailingSlash(currentPathname);
+	href = formatPath(href, {
+		format: project.build.format,
+	});
+	currentPathname =
+		project.build.format === 'file'
+			? stripTrailingSlash(currentPathname)
+			: ensureTrailingSlash(currentPathname);
+	const isCurrent = isAbsolute(href) ? false : href === currentPathname;
```

Comment:
Here on the other hand it should be safe to skip the `isAbsolute()` check, as absolute URLs will never match the pathname anyway:

```suggestion
	const isCurrent = href === currentPathname;
```